### PR TITLE
pref(bridge-history): add cache in APIs and merge RDS queries

### PR DESCRIPTION
**Pull request by colinlyguo**
### Purpose or design rationale of this PR

This PR:
1. Add cache in APIs and merge RDS queries of the same table.
2. Comment out some unused APIs and related functions.
3. Remove pagination in `GetAllClaimableTxsByAddr`.

improvement:
`GetAllClaimableTxsByAddr:`
before:
`[GIN] 2023/09/25 - 18:06:46 | 200 | 12.413356667s |       127.0.0.1 | GET      "/api/claimable?address=0xCa266224613396A0e8D4C2497DBc4F33dD6CDEFf"`
after:
`[GIN] 2023/09/25 - 18:18:31 | 200 |  4.253548958s |       127.0.0.1 | GET      "/api/claimable?address=0xCa266224613396A0e8D4C2497DBc4F33dD6CDEFf"`

### PR title

Your PR title must follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) (as we are doing squash merge for each PR), so it must start with one of the following [types](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#type):

- [x] perf: A code change that improves performance

### Deployment tag versioning

Has `tag` in `common/version.go` been updated or have you added `bump-version` label to this PR?

- [ ] No, this PR doesn't involve a new deployment, git tag, docker image tag
- [x] Yes

### Breaking change label

Does this PR have the `breaking-change` label?

- [x] No, this PR is not a breaking change
- [ ] Yes

### CODE REVIEW

File: bridge-history-api/internal/controller/history_controller.go [hunk 1]
```diff
@@ -63,10 +82,38 @@ func (c *HistoryController) PostQueryTxsByHash(ctx *gin.Context) {
 		types.RenderFailure(ctx, types.ErrParameterInvalidNo, err)
 		return
 	}
-	result, err := c.historyLogic.GetTxsByHashes(ctx, req.Txs)
-	if err != nil {
-		types.RenderFailure(ctx, types.ErrGetTxsByHashFailure, err)
-		return
+
+	const cacheKeyPrefix = "queryTxsByHash:"
```
Comment (mask-pp) [inline bridge-history-api/internal/controller/history_controller.go]: Move out from this function.

File: bridge-history-api/internal/controller/history_controller.go [hunk 2]
```diff
@@ -28,33 +35,45 @@ func (c *HistoryController) GetAllClaimableTxsByAddr(ctx *gin.Context) {
 		types.RenderFailure(ctx, types.ErrParameterInvalidNo, err)
 		return
 	}
-	offset := (req.Page - 1) * req.PageSize
-	limit := req.PageSize
-	txs, total, err := c.historyLogic.GetClaimableTxsByAddress(ctx, common.HexToAddress(req.Address), offset, limit)
+
+	cacheKey := "claimableTxsByAddr:" + req.Address
```
Comment (mask-pp) [inline bridge-history-api/internal/controller/history_controller.go]: Move cacheKey out from this function as a const value.

File: bridge-history-api/internal/controller/history_controller.go (hunk 1 above)
Comment (colinlyguo) [reply bridge-history-api/internal/controller/history_controller.go]: changed in [850be1e](https://github.com/scroll-tech/scroll/pull/966/commits/850be1e6667441fc8f118dac43603f14a97fd691)

File: bridge-history-api/internal/controller/history_controller.go (hunk 2 above)
Comment (colinlyguo) [reply bridge-history-api/internal/controller/history_controller.go]: changed in [850be1e](https://github.com/scroll-tech/scroll/pull/966/commits/850be1e6667441fc8f118dac43603f14a97fd691)

**Review by mask-pp — DISMISSED** [review]

File: bridge-history-api/internal/controller/history_controller.go [hunk 3]
```diff
@@ -63,10 +88,53 @@ func (c *HistoryController) PostQueryTxsByHash(ctx *gin.Context) {
...
-		types.RenderFailure(ctx, types.ErrGetTxsByHashFailure, err)
+
+	if len(req.Txs) > 10 {
+		types.RenderFailure(ctx, types.ErrParameterInvalidNo, errors.New("the number of hashes in the request exceeds the allowed maximum"))
 		return
 	}
-	types.RenderSuccess(ctx, &types.ResultData{Result: result, Total: 0})
+
+	// deduplicate
+	hashesMap := make(map[string]struct{})
```
Comment (mask-pp) [inline bridge-history-api/internal/controller/history_controller.go]:
```suggestion
	hashesMap := make(map[string]struct{}, len(req.Txs))
```

File: bridge-history-api/internal/controller/history_controller.go [hunk 4]
```diff
@@ -63,10 +103,66 @@ func (c *HistoryController) PostQueryTxsByHash(ctx *gin.Context) {
...
+	for _, hash := range req.Txs {
+		hashesMap[hash] = struct{}{}
+	}
+
+	txHashes := make([]string, 0, len(hashesMap))
+	for hash := range hashesMap {
```
Comment (mask-pp) [inline bridge-history-api/internal/controller/history_controller.go]: Don't need to use this loop, filter tx_hash when `range req.Txs` can be more simple.

Comment (colinlyguo) [reply bridge-history-api/internal/controller/history_controller.go]: changed in [8b93d0e](https://github.com/scroll-tech/scroll/pull/966/commits/8b93d0ea762262922636710ded81bd8fa653f5da)

File: bridge-history-api/internal/controller/history_controller.go (hunk 3 above)
Comment (colinlyguo) [reply bridge-history-api/internal/controller/history_controller.go]: changed in [8b93d0e](https://github.com/scroll-tech/scroll/pull/966/commits/8b93d0ea762262922636710ded81bd8fa653f5da)

File: bridge-history-api/internal/controller/history_controller.go [hunk 5]
```diff
@@ -63,10 +101,63 @@ func (c *HistoryController) PostQueryTxsByHash(ctx *gin.Context) {
...
+		if err != nil {
+			return
+		}
+
+		for _, result := range dbResults {
+			results = append(results, result)
+			cacheKey := cacheKeyPrefixQueryTxsByHash + result.Hash
+			c.cache.Set(cacheKey, result, cache.DefaultExpiration)
+		}
+	}
```
Comment (georgehao) [inline bridge-history-api/internal/controller/history_controller.go]:
```suggestion
	releaseChan := make(chan struct{})
	defer close(releaseChan)
	doSingleFlight := func(hashes []string) {
		for _, hash := range hashes {
			_, err, _ := c.singleFlight.Do(hash, func() (interface{}, error) {
				<-releaseChan
				return nil, nil
			})
			// add this check to fix golint.
			if err != nil {
				log.Error("unexpected error", "err", err)
			}
		}
	}

	if len(uncachedHashes) > 0 {
		go doSingleFlight(uncachedHashes)

		dbResults, err := c.historyLogic.GetTxsByHashes(ctx, uncachedHashes)
		if err != nil {
			return
		}

		for _, result := range dbResults {
			results = append(results, result)
			cacheKey := cacheKeyPrefixQueryTxsByHash + result.Hash
			c.cache.Set(cacheKey, result, cache.DefaultExpiration)
		}
	}
```

**Review by mask-pp — DISMISSED** [review]

**Review by georgehao — APPROVED** [review]

**Review by mask-pp — APPROVED** [review]

**Pull request merged**